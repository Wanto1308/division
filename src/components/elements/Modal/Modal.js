import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import { AppContext } from '../../../contexts';
import styles from './styles.scoped.css';

export default function Modal(props) {
  const { setModal, setOnCloseModal } = useContext(AppContext);
  const { open, className, children, onClose } = props;

  useEffect(() => () => setModal(null), []);

  useEffect(() => {
    setModal(open ? <Content children={children} className={className} onClose={onClose} /> : null);
    open && setOnCloseModal(() => onClose);
  }, [open]);

  useEffect(() => {
    open && setOnCloseModal(() => onClose);
  }, [onClose]);

  useEffect(() => {
    open && setModal(<Content children={children} className={className} onClose={onClose} />);
  }, [children, className]);

  return null;
}

Modal.defaultProps = {
  children: null,
  className: '',
  onClose: () => { },
  open: false,
};

Modal.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClose: PropTypes.func,
  open: PropTypes.bool,
};

export function Content({ className, children, onClose }) {
  const customClass = [styles.root, className].filter(Boolean).join(' ');

  return (
    <section className={customClass}>
      <img alt="close" onClick={onClose} src="/assets/ic-close.svg" />
      {children}
    </section>
  );
}

Content.defaultProps = {
  children: null,
  className: '',
  onClose: null,
};

Content.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClose: PropTypes.func,
};
