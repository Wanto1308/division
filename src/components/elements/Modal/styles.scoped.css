.root {
  background-color: var(--white-color);
  border-radius: 0.25rem;
  margin: auto;
  max-height: 90%;
  max-width: 90%;
  overflow-y: auto;
  padding: 3.125rem 2rem 0;
  position: relative;
  top: 50%;
  transform: translateY(-50%);
  width: fit-content;
  pointer-events: auto;
}

.root > img:first-child {
  cursor: pointer;
  position: absolute;
  right: 1rem;
  top: 1rem;
  width: 1.5rem;
}

.root::after {
  content: "";
  display: block;
  padding-bottom: 2rem;
}
