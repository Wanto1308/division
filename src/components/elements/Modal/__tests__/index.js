import React, { useContext, useEffect } from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Modal, { Content } from '../Modal';

const setModal = jest.fn();
const setOnCloseModal = jest.fn();
useContext.mockImplementation(() => ({ setModal, setOnCloseModal }));

describe('src/components/elements/Modal', () => {
  test('render', () => {
    useEffect.mockImplementationOnce(f => f()());

    Modal(Modal.defaultProps);
    Modal.defaultProps.onClose();
    expect(setModal).toHaveBeenCalledWith(null);

    const onClose = jest.fn();
    Modal({ ...Modal.defaultProps, onClose, open: true });
    expect(setModal).toHaveBeenCalledWith(<Content onClose={onClose} />);

  });

  test('Content', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Content />);
    expect(tree).toMatchSnapshot();
  });
});
