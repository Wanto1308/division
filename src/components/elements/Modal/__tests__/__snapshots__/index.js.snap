// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Modal Content 1`] = `
<section
  className=""
>
  <img
    alt="close"
    onClick={null}
    src="/assets/ic-close.svg"
  />
</section>
`;
